app/models/ssh_key.rb

```diff
@@ -0,0 +1,66 @@
+class SshKey < ActiveRecord::Base
+  include Authorizable
+  extend FriendlyId
+  friendly_id :name
+  include Parameterizable::ByIdName
+
+  belongs_to :user
+  before_validation :generate_fingerprint
+  before_validation :calculate_length
+
+  audited :associated_with => :user
+
+  scoped_search :on => :name
+  scoped_search :on => :user_id, :complete_enabled => false, :only_explicit => true, :validator => ScopedSearch::Validators::INTEGER
+
+  validates_lengths_from_database
+
+  validates :name, :user_id,
+    :presence => true
+
+  validates :key,
+    :presence => true,
+    :ssh_key => true,
+    :format => { with: /\A(ssh|ecdsa)-.*\Z/ }
+
+  validates :key,
+    :format => { :without => /\n|\r/, :message => N_('should be a single line') }
+
+  validates :fingerprint,
+    :uniqueness => true,
+    :presence => { :message => N_('could not be generated') }
+
+  validates :length,
+    :presence => { :message => N_('could not be calculated') }
+
+  delegate :name, to: :user, prefix: true
+
+  class Jail < ::Safemode::Jail
+    allow :name, :user, :key, :to_export, :fingerprint, :length
+  end
+
+  def to_export
+    type, base64 = self.key.split
+    [type, base64, user_name].join(' ')
+  end
+
+  private
+
+  def generate_fingerprint
+    self.fingerprint = nil
+    return unless self.key.present?
+    self.fingerprint = SSHKey.fingerprint(self.key)
+    true
+  rescue SSHKey::PublicKeyError
+    nil
```

Comment:
Wouldn't it be better to display a general error if this happens as `errors.add(:base => "Could not generate fingerprint - #{e})`? 

As it stands right now if this exception happens, the user cannot know what happened, or how to fix it - not even the Administrator as it does not log anything.